soroban-sdk/src/set.rs

```diff
@@ -0,0 +1,276 @@
+use core::{cmp::Ordering, fmt::Debug};
+
+use super::{Env, IntoVal, Map, RawVal, TryFromVal};
+
+/// Create a [Set] with the given items.
+///
+/// The first argument in the list must be a reference to an [Env], then the
+/// items follow.
+///
+/// ### Examples
+///
+/// ```
+/// use soroban_sdk::{Env, set};
+///
+/// let env = Env::default();
+/// let set = set![&env, 0, 1, 2, 3, 3];
+/// assert_eq!(set.len(), 4);
+/// ```
+#[macro_export]
+macro_rules! set {
+    ($env:expr) => {
+        $crate::Set::new($env)
+    };
+    ($env:expr, $($x:expr),+ $(,)?) => {
+        $crate::Set::from_array($env, [$($x), +])
+    };
+}
+
+/// Set is a contiguous growable array type composed of unique items.
```

Comment:
It's not contiguous.
```suggestion
/// Set is an ordered growable collection composed of unique items.
```

Reply:
I went back and forth on whether I should use the `ordered` word here. When I think of something **ordered**, I think that "the order in which I add items matters". And I would think `set(1, 2, 3) != set(2, 1, 3)`. 

I tried to be more specific about this implicit ordering a few lines below in the doc:

> Set imposes a fixed ordering, so Set<1, 2, 3> == Set<2, 3, 1>

What do you think of removing the `ordered` qualifier here and just say:

```suggestion
/// Set is a growable collection composed of unique items.
```
?

Reply:
Good point, it's so ambiguous to what it actually means. +1 to removing the word ordered here.